### ISSUE TITLE
insert postgres calculations into Klipfolio

### ISSUE THREAD

**Issue by btbonval** [medic/medic-webapp#1383]
Beforehand, impact metrics and other calculations that did not go into dashboards were directly reviewed in Google Spreadsheets.

Since postgres is only accessible through an SSH tunnel, the sort of people who are comfortable with Google Spreadsheets will probably be uncomfortable connecting to postgres using our technical doohickery.

Impact metrics and friends need to be put somewhere accessible, so might as well go with Klipfolio.

**Comment by btbonval** [comment]
We might also use Tableau for some non-dashboard visualizations and calculations. I was just given a key and should consider Tableau's interface as a possible replacement for Klipfolio wrt impact.

**Comment by btbonval** [comment]
* add "weeks since lmp" = # to `ancview_base`
* configuration variable for "upcoming delivery"
* add "upcoming delivery" to `ancview_delivery`
* figure out how to handle "no visits" for visits completed so far
* for visits completed so far: need to correlate active patients with visits so far

**Comment by btbonval** [comment]
Need to extract messages for nearly missed visits.

It looks like any `data_record` that has no `form` property is a message, however that might be only part of the story. It looks like messages might also be embedded into forms under `scheduled_tasks`.

```
SELECT * FROM jsonb_each( (SELECT data->'scheduled_tasks' FROM nosql) );
ERROR:  more than one row returned by a subquery used as an expression
```

**Comment by btbonval** [comment]
This is where we get the current message exports that are used in production systems:
https://github.com/medic/medic-webapp/blob/master/packages/kujua-sms/kujua-sms/lists.js#L162

**Comment by btbonval** [comment]
nearly missed visits requires messages: #1382

**Comment by btbonval** [comment]
Lost track of this ticket. I came up with a workaround to parse schedules from config and apply them to all patients, so messages were no longer a blocker.

Still no solid decision on how impact folks will look at this data, but Tableau and Klipfolio both suit the need in a pinch.

As for this issue, we made a nearly complete dashboard for HealthRight ANC drawing data from postgres. Sounds like that was the aim of the issue, closing as resolved.
